ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParseUri.java

```diff
@@ -100,9 +102,18 @@ protected PubsubMessage processElement(PubsubMessage message)
           zip(GENERIC_URI_SUFFIX_ELEMENTS, uri.substring(GENERIC_URI_PREFIX.length()).split("/")));
     } else if (uri.startsWith(StubUri.PREFIX)) {
       payload = StubUri.parse(uri, attributes);
+
+      if (attributes.get("messageId") != null) {
+        // convert PubSub message ID to document ID
+        UUID documentId = UUID
+            .nameUUIDFromBytes(attributes.get("messageId").getBytes(StandardCharsets.UTF_8));
```

Comment:
This is the first time I've encountered `nameUUIDFromBytes` and it looks like there's some need to document the specific approach here. The RFC for UUIDs discusses creating a "name space" by using a static UUID. It looks like the `nameUUIDFromBytes` expects you to prepend that namespace to the bytes you pass in; otherwise, you're assuming a null namespace. See https://stackoverflow.com/a/55296637

I think it's probably fine to use the call as written here, but we should document that it's a V3 UUID using a null namespace so that it's more obvious how you would implement the same calculation in a different language if needed.